src/Rector/Class_/PreferPHPUnitSelfCallRector.php

```diff
@@ -95,8 +99,21 @@ public function refactor(Node $node): ?Node
                 return null;
             }
 
-            $hasChanged = true;
-            return $this->nodeFactory->createStaticCall('self', $methodName, $node->getArgs());
+            if (! $this->nodeTypeResolver->getType($node->var) instanceof TypeWithClassName) {
+                return null;
+            }
+
+            $classReflection = $this->reflectionResolver->resolveClassReflectionSourceObject($node);
```

Comment:
as it from `$this` variable which already validated early; the `resolveClassReflection()` 

```php

$classReflection = $this->reflectionResolver->resolveClassReflectionSourceObject($node);
```

https://github.com/rectorphp/rector-src/blob/07a61c18f678a24c102a83a4498265c874fab617/src/Reflection/ReflectionResolver.php#L66

Reply:
What does that mean? I do not understand.

Reply:
```suggestion
            $classReflection = $this->reflectionResolver->resolveClassReflection($node);